ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/IntroEligibilityStateView.kt

```diff
@@ -0,0 +1,61 @@
+package com.revenuecat.purchases.ui.revenuecatui.composables
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+
+internal enum class IntroOfferEligibility {
+    INELIGIBLE,
+    ELIGIBLE,
+}
+
+/**
+ * A composable that can display different data based on intro eligibility status.
+ */
+@Composable
+internal fun IntroEligibilityStateView(
+    textWithNoIntroOffer: String?,
+    textWithIntroOffer: String?,
+    eligibility: IntroOfferEligibility,
+    color: Color,
+) {
+    val text: String = if (textWithIntroOffer != null && eligibility == IntroOfferEligibility.ELIGIBLE) {
+        textWithIntroOffer
+    } else {
+        // Display text with intro offer as a backup to ensure layout does not change
+        // when switching states.
+        textWithNoIntroOffer ?: textWithIntroOffer ?: ""
+    }
+
+    Text(text, color = color)
+}
+
+@Preview(showBackground = true)
+@Composable
+private fun IntroEligibilityPreview() {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
```

Comment:
Should we make this a column with 3 views or just create 3 preview functions? Seems like these previews should be independent from each other.

Reply:
Oh I see that Android studio can show multiple previews at the same time unlike Xcode. I'll change that then 👍🏻